lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -83,6 +83,25 @@ def onReplace(self, *args):
         # Nothing needed here since we don't create any UI.
         pass
 
+class UfeConnectionChangedObserver(ufe.Observer):
+    def __init__(self, item):
+        super(UfeConnectionChangedObserver, self).__init__()
+        self._item = item
+
+    def __del__(self):
```

Comment:
Are you sure this is necessary?  Observers are kept by weak ptr by their Subject(s), so if the shared ptr goes away, they will be cleaned up by all Subject(s) they were added to.

Reply:
No idea. I simply replicated the structure of UfeAttributesObserver found directly above this class, including the __del__ found at line 63 of this file.
Either both are correct, or both are wrong. Want me to check?
I do have to agree that relying on the Python garbage collector to remove observers does not look like a good long term strategy.

Reply:
Yeah, that would be a useful exercise.  We can document your findings in the code, wherever is most useful.

Reply:
After a bit of investigation, I would say the current code works very well and it is always a good idea to deregister a listener. Would be nice to be able to deregister as soon as the AE page gets replaced because I could find a very small window where the listener was still active even though the AE was showing another page. But that does not worry me since the garbage collector is very aggressive, the window is small, and the effect is only an unexpected AE refresh.

Reply:
The reason for this is because the AE actually has multiple tabs - one for each object type. So if you switch between a USD Cone and USD Sphere, the AE is actually hiding one tab and showing the other. All the controls on the hidden tab are left behind. This is because of the "replace" mode where Maya will reuse the existing controls for the next object of the type same.